src/app/search/search-utils.ts

```diff
@@ -1,17 +1,48 @@
-import { parseQuery, QueryAST } from './query-parser';
+import { canonicalizeQuery, parseQuery, QueryAST } from './query-parser';
 import { SearchConfig } from './search-config';
 
-export function parseAndValidateQuery(query: string, searchConfig: SearchConfig) {
+export function parseAndValidateQuery(
+  query: string,
+  searchConfig: SearchConfig
+): {
+  /** Is the query valid at all? */
+  valid: boolean;
+  /** Can the user save this query? */
+  saveable: boolean;
+  /** Should we automatically save this in search history? */
+  saveInHistory: boolean;
+  /** The canonicalized version of the query */
+  canonical: string;
+} {
   let valid = true;
+  let saveable = true;
+  let saveInHistory = true;
+  let canonical = query;
   try {
     const ast = parseQuery(query);
     if (!validateQuery(ast, searchConfig)) {
       valid = false;
+    } else {
+      if (ast.op === 'noop' || (ast.op === 'filter' && ast.type === 'keyword')) {
+        // don't save "trivial" single-keyword filters
+        saveInHistory = false;
+      }
+      // Some sites have people save big lists of item IDs. Even if these aren't too long, don't save them automatically
+      if (ast.op === 'or' && ast.operands.every((op) => op.op === 'filter' && op.type === 'id')) {
+        saveInHistory = false;
```

Comment:
This does however save singular `id:` queries in the history.

Reply:
True, I'm mostly just trying to avoid the big searches generated by tools.